docs/content/release-notes/flink-2.0.md

```diff
@@ -0,0 +1,1666 @@
+
+We have also upgraded Kryo to version 5.6, which is faster and more memory efficient, and has better supports for newer Java versions.
+
+# Highlight Breaking Changes
+
+## API
+
+The following sets of APIs have been completely removed.
+- **DataSet API** - Please migrate to DataStream API, or Table API/SQL if applicable. See also "How to Migrate from DataSet to DataStream".
+- **Scala DataStream and DataSet API** - Please migrate to the Java DataStream API.
+- **SourceFuction, SinkFunction and Sink V1** - Please migrate to Source and Sink V2.
+- **TableSoure and TableSink** - Please migrate to DynamicTableSource and DynamicTableSink. See also "User-defined Sources & Sinks".
+- **TableSchema, TableColumn and Types** - Please migrate to Schema, Column and DataTypes respectively.
+
+Some deprecated methods have been removed from DataStream API. See also the list of breaking programming APIs.
+
+Some deprecated fields have been removed from REST API. See also the list of breaking REST APIs.
+
+**NOTICE:** You may find some of the removed APIs still exist in the code base, usually in a different package. They are for internal usages only and can be changed / removed anytime without notifications. Please **DO NOT USE** them.
+
+### Connector Adaption Plan
+
+As SourceFunction, SinkFunction and SinkV1 being removed, existing connectors depending on these APIs will not work on the Flink 2.x series. Here’s the plan for adapting the first-party connectors.
+- A new version of Kafka, Paimon, JDBC and ElasticSearch connectors, adapted to the API changes, will be released right after the release of Flink 2.0.0.
+- We plan to gradually migrate the remaining first-party connectors within 3 subsequent minor releases (i.e., by Flink 2.3).
+
+## Configuration
+
+Configuration options meet the following criteria are removed. See also the list of removed configuration options.
+- Annotated as `@Public` and have been deprecated for at least 2 minor releases.
+- Annotated as `@PublicEvolving` and have been deprecated for at least 1 minor releases.
+
+The legacy configuration file `flink-conf.yaml` is no longer supported. Please use `config.yaml` with standard YAML format instead. A migration tool is provided to convert a legacy `flink-conf.yaml` into a new `config.yaml`. See "Migrate from flink-conf.yaml to config.yaml" for more details.
+
+Configuration APIs that takes java objects as arguments are removed from `StreamExecutionEnvironment` and `ExecutionConfig`. They should now be set via `Configuration` and `ConfigOption`. See also the list of breaking programming APIs.
+
+To avoid exposing internal interfaces, User-Defined Functions no longer have full access to `ExecutionConfig`. Instead, necessary functions such as `createSerializer()`, `getGlobalJobParameters()` and `isObjectReuseEnabled()` can now be accessed from `RuntimeContext` directly.
+
+## Misc
+
+- State Compatibility is not guaranteed between 1.x and 2.x.
```

Comment:
I would like to see 
- more details on what definitely does not work, for example Scala.
- what does work ( for example state stored in the canonical form?) 
- a list of migration considerations for example
     - identifying state that will not migrate and what the consequence is for those jobs. 
     - if the state is not compatible, what do we recommend. How do we smooth the upgrade? Is it always stop the job and rewrite losing all existing state?

Reply:
State compatibility issues are difficult to describe in a nutshell(To be honest, I don't know which states are definitely incompatible), and I think we need time to put together a migration manual/guide. Given the 2.0 release timeline, this can be done after release.

A related topic is, can we provide a state migration tool to help users migrate across incompatible checkpoints / savepoints? I.e., deserialize the states into in-memory objects with old serialization setups, and serialize them with new setups.

Reply:
@reswqa ok - if we take this approach, I think we should mention this in the release notes and have a link to the Jira that tracks this work. WDYT?